08-CMSIS/core/src/threads.c

```diff
@@ -45,20 +45,38 @@ void __attribute__((naked)) pendsv_handler()
 }
 
 
-void thread_start()
+void __attribute__((naked)) thread_start()
 {
 	lastTask = 0;
+	CONTROL_Type user_ctx = {
+		.b.nPRIV = 1,
+		.b.SPSEL = 1
+	};
+
+	/* Reset APSR before context switch */
+	asm volatile("mov r0, #0\n"
+	             "msr APSR_nzcvq, r0\n");
+	/* Load user task's context and jump to the task */
+	__set_PSP((uint32_t) tasks[lastTask].stack);
+	__set_CONTROL(user_ctx.w);
+	__ISB();
+
+	/* This is how we simulate stack handling that pendsv_handler
+	 * does. Thread_create sets 17 entries in stack, and the 9
+	 * entries we pop here will be pushed back in pendsv_handler
+	 * in the same order.
+	 */
+	asm volatile("pop {r4-r11, lr}\n"
+	             "ldr r0, [sp]\n");
+	/* Okay, we are ready to run test1, get test1 address from
```

Comment:
Avoid mentioing `test1` since it is not really specific to the whole design. Instead, use general terms.